desktop/CMakeLists.txt

```diff
@@ -20,12 +20,40 @@ set(qtkeychain_INCLUDE_DIR ${qtkeychain_BUILDDIR}/include/qt5keychain)
 
 ExternalProject_Add(qtkeychain
   GIT_REPOSITORY https://github.com/status-im/qtkeychain.git
-  CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${qtkeychain_BUILDDIR} -DQTKEYCHAIN_STATIC=ON -DBUILD_TRANSLATIONS=OFF
+  CMAKE_ARGS "-DCMAKE_INSTALL_PREFIX=${qtkeychain_BUILDDIR}"
+             "-DQTKEYCHAIN_STATIC=ON"
+             "-DBUILD_TRANSLATIONS=OFF"
+             "-DCMAKE_C_FLAGS=${CMAKE_C_FLAGS}"
+             "-DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}"
+             "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}"
+             "-DBUILD_SHARED_LIBS=${BUILD_SHARED_LIBS}"
+             # These are only useful if you're cross-compiling.
+             # They, however, will not hurt regardless.
+             "-DCMAKE_SYSTEM_NAME=${CMAKE_SYSTEM_NAME}"
```

Comment:
@PombeirP maybe it's possible to embrace all these additional CMAKE flags with check for the platform, like it's done below `if (WIN32)` ?

```
 set (KEYCHAIN_CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${qtkeychain_BUILDDIR} -DQTKEYCHAIN_STATIC=ON -DBUILD_TRANSLATIONS=OFF)
if (WIN32)
    set (KEYCHAIN_CMAKE_ARGS ${KEYCHAIN_CMAKE_ARGS } -DCMAKE_SYSTEM_NAME=${CMAKE_SYSTEM_NAME}
            ....
            )
```

Reply:
Makes sense, I'll do that. Thanks!

Reply:
@MaxRis I was thinking about this and I don't think it makes sense to make separate logic for a specific platform. These flags are meant to pass on the build context to the external project, and it doesn't really matter that we're building WIN32 or something else. It's probably even good practice to use the same code path for all platforms. That doesn't preclude us from separating them into another array and then do something like `CMAKE_ARGS ${KEYCHAIN_CMAKE_ARGS} ${PARENT_CMAKE_ARGS}`. What do you think?

Reply:
sounds good, hopefully, all these flags set explicitly will not break available builds

Reply:
Yeah, we'll need to test for that before merging this.